open_optical_gating/cli/prospective_optical_gating.py

```diff
@@ -46,23 +59,32 @@ def update_drift(frame0, bestMatch0, settings):
         frames[counter, :, :] = frame0[rectF[0] : rectF[1], rectF[2] : rectF[3]]
         counter = counter + 1
 
+    # Compare all these candidate shifted images against the matching reference frame, and find the best-matching shift
     sad = jps.sad_with_references(bestMatch, frames)
     best = np.argmin(sad)
 
-    settings["drift"][0] = dx + shifts[best][0]
-    settings["drift"][1] = dy + shifts[best][1]
+    settings["drift"][0] = dx + candidateShifts[best][0]
+    settings["drift"][1] = dy + candidateShifts[best][1]
 
     return settings
 
 
 def subframe_fitting(diffs, settings):
-    # Find the sub-pixel position fit from a list of SADs
-    # (including padding frames)
-    # Initialise best and worst scores based on padding frames
+    """ Identify the location of the minimum value (to sub-frame accuracy) within a list of diffs (including padding frames).
+        The aim is to find the position of the minimum within the "main" frames,
+        but we may have to extend to the padding frames if the minimum is right at one end of the "main" frames.
+        
+        Parameters:
+            diffs       list/array-like  Sequence of recently-received frame pixel arrays (in chronological order)
+            settings    dict             Parameters controlling the sync algorithms
+        Returns:
+            float coordinate for location of minimum in 'diffs' (including padding frames)
+        """
     bestScore = diffs[settings["numExtraRefFrames"]]
     bestScorePos = settings["numExtraRefFrames"]
 
-    # Search for lowest V
+    # Search for lowest value within the "main" frames.
+    # TODO: not a real performance issue, but this can easily be rewritten loop-free using np.argmax etc.
```

Comment:
Lol! This is funny. Why did I not use numpy?!

Reply:
I think that's what happens when you are translating C code line-by line!